torchci/components/benchmark/tritonbench/TimeSeries.tsx

```diff
@@ -0,0 +1,211 @@
+            queryName={queryName}
+            queryParams={queryParams}
+            granularity={granularity}
+            repo={repo}
+            suite={suite}
+            metric_name={metric_name}
+            branch={branch}
+            lCommit={lCommit}
+            rCommit={rCommit}
+          />
+        </CardContent>
+      </GraphCardGroup>
+    </>
+  );
+}
+
+const GRAPH_ROW_HEIGHT = 245;
+
+function SingleGraphPanel({
+  queryName,
+  queryParams,
+  granularity,
+  repo,
+  suite,
+  metric_name,
+  branch,
+  lCommit,
+  rCommit,
+}: {
+  queryName: string;
+  queryParams: { [key: string]: any };
+  granularity: Granularity;
+  repo: string;
+  suite: string;
+  metric_name: string;
+  branch: string;
+  lCommit: string;
+  rCommit: string;
+}) {
+  const queryParamsWithSuite: { [key: string]: any } = {
+    ...queryParams,
+    repo: repo,
+    suite: suite,
+    metric_name: metric_name,
+    branch: branch,
+  };
+  // NB: Querying data for all the suites blows up the response from the database
+  // over the lambda reponse body limit of 6MB. So I need to split up the query
+  // here into multiple smaller ones to keep them under the limit
+  //
+  // See more:
+  // * https://nextjs.org/docs/messages/api-routes-body-size-limit
+  // * https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html
+  const url = `/api/clickhouse/${queryName}?parameters=${encodeURIComponent(
+    JSON.stringify(queryParamsWithSuite)
+  )}`;
+
+  let { data, error } = useSWR(url, fetcher, {
+    refreshInterval: 60 * 60 * 1000, // refresh every hour
+  });
+
+  if (error !== undefined) {
+    return (
+      <div>
+        An error occurred while fetching data, perhaps there are too many
+        results with your choice of time range and granularity?
+      </div>
+    );
+  }
+
+  if (data === undefined || data.length === 0) {
+    return <Skeleton variant={"rectangular"} height={"100%"} />;
+  }
+
+  const groupByFieldName = undefined;
+  // Only show records between these twos
+  const lWorkflowId = COMMIT_TO_WORKFLOW_ID[lCommit];
+  const rWorkflowId = COMMIT_TO_WORKFLOW_ID[rCommit];
+
+  const compileTime = computeCompileTime(data).filter((r: any) => {
+    const id = r.workflow_id;
+    return (
+      (id >= lWorkflowId && id <= rWorkflowId) ||
+      (id <= lWorkflowId && id >= rWorkflowId)
+    );
+  });
+
+  // Clamp to the nearest granularity (e.g. nearest hour) so that the times will
+  // align with the data we get from the database
+  const startTime = dayjs(queryParams["startTime"]).startOf(granularity);
+  const stopTime = dayjs(queryParams["stopTime"]).startOf(granularity);
+
+  const compileTimeSeries = seriesWithInterpolatedTimes(
+    compileTime,
+    startTime,
+    stopTime,
+    granularity,
+    groupByFieldName,
+    TIME_FIELD_NAME,
+    "compilation_latency",
+    false
+  );
+
+  return (
+    <>
+      <Grid2 container spacing={2}>
+        <Grid2 size={{ xs: 12, lg: 6 }} height={GRAPH_ROW_HEIGHT}>
```

Comment:
If there is only one chart, you could set `lg` to 12 to display it in the whole horizontal space instead of half the screen I think

Reply:
We will start from one chart and add more subgraphs later, e.g., more workload suites and break-down of the compilation time by stages (e.g., TTIR, TTGIR compile time).